compiler/rustc_metadata/src/locator.rs

```diff
@@ -538,7 +539,16 @@ impl<'a> CrateLocator<'a> {
                             continue;
                         }
                     }
-                    Err(err) => {
+                    Err(MetadataError::LoadFailure(err)) => {
+                        // The file was present and created by the same compiler version, but we
+                        // couldn't load it for some reason.  Give a hard error instead of silently
+                        // ignoring it, but only if we would have given an error anyway.
```

Comment:
Can you keep the `warn!()` in this case? It may still be useful.

Reply:
Sure, done.